zookeeper-server/src/test/java/org/apache/zookeeper/server/auth/znode/groupacl/TestFixupACL.java

```diff
@@ -29,59 +30,123 @@
+
+  private List<ACL> aclList = new ArrayList<>();
+
+  {
+    aclList
+        .add(new ACL(ZooDefs.Perms.ALL, new Id(X509AuthenticationUtil.X509_SCHEME, "toBeAdded")));
+    aclList.add(new ACL(ZooDefs.Perms.ALL, ZooDefs.Ids.ANYONE_ID_UNSAFE));
+  }
+
+  private final Map<String, String> SYSTEM_PROPERTIES = new HashMap<>();
 
-  static {
+  {
+    SYSTEM_PROPERTIES
+        .put("zookeeper.authProvider.x509", X509ZNodeGroupAclProvider.class.getCanonicalName());
     SYSTEM_PROPERTIES.put(X509AuthenticationConfig.SET_X509_CLIENT_ID_AS_ACL, "true");
     SYSTEM_PROPERTIES
         .put(X509AuthenticationConfig.ZOOKEEPER_ZNODEGROUPACL_SUPERUSER_ID, "superUserId");
-    SYSTEM_PROPERTIES.put(X509AuthenticationConfig.CROSS_DOMAIN_ACCESS_DOMAIN_NAME, "crossDomain");
   }
 
-  @BeforeClass
-  public static void beforeClass() {
+  @Before
+  public void before() {
     SYSTEM_PROPERTIES.forEach(System::setProperty);
   }
 
-  @AfterClass
-  public static void afterClass() {
+  @After
+  public void after() {
     SYSTEM_PROPERTIES.keySet().forEach(System::clearProperty);
+    X509AuthenticationConfig.reset();
   }
 
   @Test
   public void testCrossDomainComponents() throws KeeperException.InvalidACLException {
     List<ACL> returnedList =
         PrepRequestProcessor.fixupACL(testPath, crossDomainComponentAuthInfo, aclList);
     Assert.assertEquals(1, returnedList.size());
-    Id returnedId = returnedList.get(0).getId();
-    Assert.assertEquals("x509", returnedId.getScheme());
-    Assert.assertEquals("crossDomain", returnedId.getId());
+    Assert.assertTrue(returnedList.contains(
+        new ACL(ZooDefs.Perms.ALL, new Id(X509AuthenticationUtil.X509_SCHEME, crossDomain))));
   }
 
   @Test
   public void testSuperUserId() throws KeeperException.InvalidACLException {
-    List<ACL> returnedList =
-        PrepRequestProcessor.fixupACL(testPath, superUserAuthInfo, aclList);
+    List<ACL> returnedList = PrepRequestProcessor.fixupACL(testPath, superUserAuthInfo, aclList);
+    Assert.assertEquals(2, returnedList.size());
+    Assert.assertTrue(returnedList.containsAll(aclList));
+  }
+
+  @Test
+  public void testSingleDomainUser() throws KeeperException.InvalidACLException {
+    List<ACL> returnedList = PrepRequestProcessor.fixupACL(testPath, singleDomainAuthInfo, aclList);
     Assert.assertEquals(1, returnedList.size());
-    Id returnedId = returnedList.get(0).getId();
-    Assert.assertEquals("x509", returnedId.getScheme());
-    Assert.assertEquals("toBeAdded", returnedId.getId());
+    Assert
+        .assertTrue(returnedList.contains(new ACL(ZooDefs.Perms.ALL, singleDomainAuthInfo.get(0))));
+  }
+
+  @Test
+  public void testCnxnFiltering() throws KeeperException.InvalidACLException {
+    System.setProperty(X509AuthenticationConfig.DEDICATED_DOMAIN, dedicatedDomain);
+    List<ACL> returnedList =
+        PrepRequestProcessor.fixupACL(testPath, dedicatedDomainAuthInfo, aclList);
+    Assert.assertEquals(2, returnedList.size());
```

Comment:
Nit: could we explain what we're doing and what the expected values should be in the `returnedList` for other developers who'll get to work on this in the future?